### PR TITLE
Card12

### PR DESCRIPTION
**Pull request by Radon-Rodion**
Changes in redux and age filtration + card 12 done

### CODE REVIEW

File: src/redux/store/reducers/cartReducer.ts [hunk 1]
```diff
@@ -22,7 +19,7 @@ const cartReducer = (state = defaultState, action: CartAction): Cart => {
       const elementIndex = findCartElement(tempArr, newElement.name, newElement.orderDate);
       if (elementIndex === -1) tempArr.push(newElement);
       else tempArr[elementIndex].amount += 1;
-      serialize(tempArr);
+      serialize<CartElement[]>(tempArr, CART);
       return { elements: tempArr };
```
Comment (stnslvtvrn) [inline src/redux/store/reducers/cartReducer.ts]: wrong return

File: src/redux/store/reducers/cartReducer.ts [hunk 2]
```diff
@@ -10,10 +11,6 @@ const defaultState: Cart = {
   elements: JSON.parse(localStorage.getItem(CART) ?? "[]"),
 };
 
-const serialize = (elements: CartElement[]): void => {
-  localStorage.setItem(CART, JSON.stringify(elements));
-};
-
 const cartReducer = (state = defaultState, action: CartAction): Cart => {
   let tempArr = [...state.elements];
```
Comment (stnslvtvrn) [inline src/redux/store/reducers/cartReducer.ts]: bad naming

File: src/redux/store/reducers/cartReducer.ts [hunk 3]
```diff
@@ -31,12 +28,12 @@ const cartReducer = (state = defaultState, action: CartAction): Cart => {
       else if (editParams.newChosenPlatformIndex !== undefined)
         tempArr[editParams.index].chosenPlatformIndex = editParams.newChosenPlatformIndex;
       else tempArr[editParams.index].selected = !tempArr[editParams.index].selected;
-      serialize(tempArr);
+      serialize<CartElement[]>(tempArr, CART);
       return { elements: tempArr };
```
Comment (stnslvtvrn) [inline src/redux/store/reducers/cartReducer.ts]: wrong return

File: src/redux/store/reducers/cartReducer.ts [hunk 4]
```diff
@@ -31,12 +28,12 @@ const cartReducer = (state = defaultState, action: CartAction): Cart => {
...
     }
     case DELETE_SELECTED:
       tempArr = tempArr.filter((element) => !element.selected);
-      serialize(tempArr);
+      serialize<CartElement[]>(tempArr, CART);
       return { elements: tempArr };
```
Comment (stnslvtvrn) [inline src/redux/store/reducers/cartReducer.ts]: wrong return

File: src/components/gameCard/gameCard.tsx [hunk 5]
```diff
@@ -17,17 +17,23 @@ interface IGameCardProps {
...
       dispatch(addGame(props.gameInfo));
       alert("Got product");
     } else {
       alert("Sign in first!");
     }
   };
 
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const adminForm = <AdminForm formInfo={formByName("Edit card")} gameInfo={props.gameInfo} onExit={() => {}} />;
+
```
Comment (stnslvtvrn) [inline src/components/gameCard/gameCard.tsx]: empty onExit function?????? remove all useless functions. also in products, gameCard, profile

File: src/pages/products/products.tsx [hunk 6]
```diff
@@ -45,18 +40,18 @@ const [filters, setFilters] = useState({ ...defaultFilters, category: props.cate
     updateFilters(filters);
   }, [filters]);
 
+  const isAdmin = useSelector((state: RootState) => state.user.isAdmin);
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const adminForm = <AdminForm formInfo={formByName("Create card")} gameInfo={newProductInfo} onExit={() => {}} />;
+
```
Comment (stnslvtvrn) [inline src/pages/products/products.tsx]: useless function

**Review by stnslvtvrn — CHANGES_REQUESTED** [review]

File: src/redux/store/reducers/cartReducer.ts (hunk 1 above)
Comment (Radon-Rodion) [reply src/redux/store/reducers/cartReducer.ts]: reducers returns rewritten as you said

File: src/redux/store/reducers/cartReducer.ts (hunk 2 above)
Comment (Radon-Rodion) [reply src/redux/store/reducers/cartReducer.ts]: tempArr renamed into "cartElements"

File: src/redux/store/reducers/cartReducer.ts (hunk 3 above)
Comment (Radon-Rodion) [reply src/redux/store/reducers/cartReducer.ts]: reducer returns fixed

File: src/components/gameCard/gameCard.tsx (hunk 5 above)
Comment (Radon-Rodion) [reply src/components/gameCard/gameCard.tsx]: empty functions all over the project removed / replaced with 'undefined'

**Review by stnslvtvrn — APPROVED** [review]

**Pull request merged**